server/src/main.rs

```diff
@@ -60,32 +62,32 @@ fn run_server(host: &str, port: &str, repo_root: &Path) {
 }
 
 fn get_repo(req: &HttpRequest<AppState>) -> impl Responder {
-    //TODO https://actix.rs/docs/errors/
-    let path_params = actix_web::Path::<PathParams>::extract(req).expect("Wrong path params");
-    let query_params = actix_web::Query::<QueryParams>::extract(req).expect("Wrong query params");
-    let repo_key = path_params.repo.clone();
-    let filename = query_params.file.clone();
-    let reference = format!(
-        "refs/{}",
-        query_params
-            .reference
-            .as_ref()
-            .map(String::as_str)
-            .unwrap_or(DEFAULT_REFERENCE)
-    );
-    let gr: &Addr<GitRepos> = &req.state().git_repos;
-    //TODO return proper content type depending on the content of the blob
-    gr.send(CatFile {
-        repo_key,
-        filename,
-        reference,
-    })
-    .map(|x| {
-        x.0.map(Binary::from)
-            .map_err(|e| actix_web::error::InternalError::new(e, http::StatusCode::NOT_FOUND))
+    actix_web::Path::<PathParams>::extract(req).and_then(|path_params| {
+        actix_web::Query::<QueryParams>::extract(req).and_then(|query_params| {
+            let repo_key = path_params.repo.clone();
+            let filename = query_params.file.clone();
+            let reference = format!(
+                "refs/{}",
+                query_params
+                    .reference
+                    .as_ref()
+                    .unwrap_or(&DEFAULT_REFERENCE.to_string())
```

Comment:
We lost a change here to prevent the cloning of the string unnecessarily:

```suggestion
                    .map(String::as_str)
                    .unwrap_or(DEFAULT_REFERENCE)
```